experimental/robust_segvit/custom_segmentation_trainer.py

```diff
@@ -160,17 +174,51 @@ def evaluate(train_state: train_utils.TrainState,
   # Evaluate global metrics on one of the hosts (lead_host), but given
   # intermediate values collected from all hosts.
 
-  for _ in range(steps_per_eval):
+  # start ece metric
+  ece_metric = rm.metrics.ExpectedCalibrationError(num_bins=10)._metric
+
+  # store logits
+  store_logits = config.eval_configs.get('store_logits', False)
+
+  if store_logits:
+    store_logits_fname = os.path.join(workdir, "{}_{}_val.h5py".format(prefix,"logits"))
+    f = h5py.File(store_logits_fname, 'w', libver='latest')
+    f.swmr_mode = True  # single write multi-read
+    input_shape = dataset.meta_data['input_shape'][1:3]
+    num_classes = dataset.meta_data['num_classes']
+    num_eval_examples = int(steps_per_eval * config.batch_size)
+    logits_out = f.create_dataset('logits', (num_eval_examples,) + input_shape + (num_classes,))
+    inputs_out = f.create_dataset('inputs', (num_eval_examples,) + input_shape + (3,))
+    labels_out = f.create_dataset('labels', (num_eval_examples,) + input_shape)
+
+  for step_ in range(steps_per_eval):
     eval_batch = next(dataset.valid_iter)
     e_batch, e_logits, e_metrics, confusion_matrix, unc_confusion_matrix = eval_step_pmapped(
         train_state=train_state, batch=eval_batch)
     eval_metrics.append(train_utils.unreplicate_and_get(e_metrics))
+
+    probs = jax.nn.softmax(e_logits, axis=-1)
+
+    # TODO(kellybuchanan): add masking to ece metric in rm.
+    # updates on each host separately
+    ece_metric.update_state(e_batch['label'], probs, sample_weight=e_batch['batch_mask'])
```

Comment:
Based on the comment in https://github.com/google-research/robustness_metrics/pull/77, can we move the binary masking logic here such that we pass in a masked subset of labels and probs as needed without changing the ECE metric API?